src/luaotfload-notdef.lua

```diff
@@ -0,0 +1,67 @@
+-----------------------------------------------------------------------
+--         FILE:  luaotfload-notdef.lua
+--  DESCRIPTION:  part of luaotfload / notdef
+-----------------------------------------------------------------------
+
+local ProvidesLuaModule = { 
+    name          = "luaotfload-notdef",
+    version       = "2.9805",       --TAGVERSION
+    date          = "2019-06-20", --TAGDATE
+    description   = "luaotfload submodule / color",
+    license       = "GPL v2.0",
+    author        = "Marcel Krüger"
+}
+
+if luatexbase and luatexbase.provides_module then
+  luatexbase.provides_module (ProvidesLuaModule)
+end  
+
+local nodenew            = node.direct.new
+local getfont            = font.getfont
+local setfont            = node.direct.setfont
+local getwhd             = node.direct.getwhd
+local insert_after       = node.direct.insert_after
+local traverse_char      = node.direct.traverse_char
+local protect_glyph      = node.direct.protect_glyph
+local otffeatures        = fonts.constructors.newfeatures "otf"
+
+local function setnotdef(tfmdata, factor)
+  tfmdata.notdefcode = tfmdata.resources.unicodes[".notdef"]
```

Comment:
I'm not sure relying on fonts having correct glyph names is a good idea. In OpenType glyph names are of secondary imporance and fonts can go away with all sorts of broken naming, so I'd not be surprised if there are fonts somewhere that names a regular glyph `.notdef`. Always using index 0 is probably a better idea for OpenType/TrueType fonts.